wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Message Details/MessageDetailsContentViewController.swift

```diff
@@ -254,28 +282,73 @@ final class MessageDetailsContentViewController: UIViewController {
 extension MessageDetailsContentViewController: UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
 
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return cells.count
+        return sections[section].items.count
+    }
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return sections.count
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        referenceSizeForHeaderInSection section: Int
+    ) -> CGSize {
+        header.titleLabel.text = "headerText"
+        header.size(fittingWidth: collectionView.bounds.width)
+        return header.bounds.size
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        viewForSupplementaryElementOfKind kind: String,
+        at indexPath: IndexPath
+    ) -> UICollectionReusableView {
+        let view = collectionView.dequeueReusableSupplementaryView(
+            ofKind: UICollectionView.elementKindSectionHeader,
+            withReuseIdentifier: "SectionHeader",
```

Comment:
I suggest to avoid duplicating this string.

Reply:
`"SectionHeader"` occurs in this file multiple times. If you change on occurrence and not the other then it will cause a crash. Therefore it's best to store the value in a constant or compute the value so if you change it in one place it will change in all places.

Reply:
Done!